LibZipSharp.UnitTest/ZipArchiveExTest.cs

```diff
@@ -0,0 +1,153 @@
+﻿using NUnit.Framework;
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using Xamarin.Tools.Zip;
+using Xamarin.Tools.Zip.Ex;
+
+namespace Tests {
+
+	public class ZipArchiveExTest {
+
+		Random rnd = new Random (234187614);
+
+		void CreateBaseFilesDirectory ()
+		{
+			if (Directory.Exists ("Resources"))
+				Directory.Delete ("Resources", recursive: true);
+			Directory.CreateDirectory ("Resources");
+			Directory.CreateDirectory (Path.Combine ("Resources", "subfolder"));
+			
+			for (int i = 0; i < 500; i++) {
+				int fileSize = rnd.Next (short.MaxValue * 4);
+				byte [] buffer = new byte [fileSize];
+				rnd.NextBytes (buffer);
+				File.WriteAllBytes (Path.Combine ("Resources", $"File{i}.txt"), buffer);
+			}
+			for (int i = 0; i < 500; i++) {
+				int fileSize = rnd.Next (short.MaxValue * 4);
+				byte [] buffer = new byte [fileSize];
+				rnd.NextBytes (buffer);
+				File.WriteAllBytes (Path.Combine ("Resources", "subfolder", $"FileSub{i}.txt"), buffer);
+			}
+		}
+
+		void CreateNewFilesDirectory ()
+		{
+			if (Directory.Exists ("NewFiles"))
+				Directory.Delete ("NewFiles", recursive: true);
+			Directory.CreateDirectory ("NewFiles");
+			Directory.CreateDirectory (Path.Combine ("NewFiles", "NewFiles1"));
+			for (int i = 0; i < 500; i++) {
+				int fileSize = rnd.Next (short.MaxValue * 4);
+				byte [] buffer = new byte [fileSize];
+				rnd.NextBytes (buffer);
+				File.WriteAllBytes (Path.Combine ("NewFiles", $"File{i}.txt"), buffer);
+			}
+			for (int i = 0; i < 500; i++) {
+				int fileSize = rnd.Next (short.MaxValue * 4);
+				byte [] buffer = new byte [fileSize];
+				rnd.NextBytes (buffer);
+				File.WriteAllBytes (Path.Combine ("NewFiles", "NewFiles1", $"FileSub{i}.txt"), buffer);
+			}
+		}
+
+		CompressionMethod GetCompressionMethod (string fileName)
+		{
+			if (fileName.Contains ("5"))
+				return CompressionMethod.Store;
+			return CompressionMethod.Deflate;
+		}
+
+		[Test]
+		public void CreateaValidZipArchive ()
```

Comment:
I was thinking this could try a `[Repeat (100)]` or something crazy. Or maybe just doing `ZipArchiveEx.Flush()` in xamarin-android?

I'll try a couple things, too, and see if I come up with something.

Reply:
I did that locally when I first wrote the test `[Repeat (1000)]`. It never failed 🤷‍♂

Reply:
I added the repeat  to see what happens on the bots